### DISCUSSION TITLE
can not install addon in offline mode such as cloudshell

### DISCUSSION THREAD

**Discussion by liutao841204** [kubevela/kubevela#4999] [Q&A] [answered]
公司服务器因为使用了网络代理，不能直接上网，所以在安装cloudshell时，出现了如下错误：
<img width="1223" alt="image" src="https://user-images.githubusercontent.com/24605302/199703212-7fee233c-81ba-4840-a05b-27fddd7a056f.png">
使用的命令：vela addon enable cloudshell

是否可以提供一个单独的helm部署方式！！！

**Comment by liutao841204** [comment]
<img width="1463" alt="image" src="https://user-images.githubusercontent.com/24605302/199703847-d9630a9c-6665-4480-999e-e3c6e51d0148.png">
我的插件列表也不能用，可以告诉我在哪里设置代理，您呢个让它上网吗？

**Comment by barnettZQG** [comment]
Refer to: https://github.com/kubevela/catalog/blob/master/addons/cloudshell/template.yaml

You can try to install it via helm.

**Comment by liutao841204** [comment]
> Refer to: https://github.com/kubevela/catalog/blob/master/addons/cloudshell/template.yaml
> 
> You can try to install it via helm.

插件列表的事您有什么建议吗？

**Comment by liutao841204** [comment]
> You can try to install it via helm.

您好，给我文件里的这个链接里也没有chart定义啊！
<img width="1070" alt="image" src="https://user-images.githubusercontent.com/24605302/199865419-f9f5564f-6566-4ac3-b2bc-118e17aa9543.png">

**Comment by barnettZQG** [comment]
@liutao841204 安装插件涉及需要网络的地方包括：

1. vela-system/vela-apiserver 请求插件列表，域名：addons.kubevela.net
2. flux-system/fluxcd-source-controller 请求 Charts 仓库，域名由具体 Charts 仓库决定。
3. 节点拉取镜像，域名由镜像名称决定。

如果可以设置代理上网，可以考虑在这三个环境设置，HTTP_PROXY 环境变量?

**Comment by liutao841204** [comment]
> @liutao841204 安装插件涉及需要网络的地方包括：
> 
> 1. vela-system/vela-apiserver 请求插件列表，域名：addons.kubevela.net
> 2. flux-system/fluxcd-source-controller 请求 Charts 仓库，域名由具体 Charts 仓库决定。
> 3. 节点拉取镜像，域名由镜像名称决定。
> 
> 如果可以设置代理上网，可以考虑在这三个环境设置，HTTP_PROXY 环境变量?
我看了vela-apiserver ，ping了
<img width="1129" alt="image" src="https://user-images.githubusercontent.com/24605302/199903591-0130f56b-fbb8-4e30-ba50-bd1fd49b946f.png">
也看了日志：
{"level":"error","ts":1667532870.0720513,"caller":"addon/cache.go:325","msg":"fail to get addons from registry KubeVela for cache updating, download index file from https://addons.kubevela.net failure Get \"https://addons.kubevela.net/index.yaml\": http: server gave HTTP response to HTTPS client","stacktrace":"github.com/oam-dev/kubevela/pkg/addon.(*Cache).listVersionRegistryUIDataAndCache\n\t/workspace/pkg/addon/cache.go:325\ngithub.com/oam-dev/kubevela/pkg/addon.(*Cache).ListUIData\n\t/workspace/pkg/addon/cache.go:142\ngithub.com/oam-dev/kubevela/pkg/apiserver/domain/service.(*addonServiceImpl).ListAddons\n\t/workspace/pkg/apiserver/domain/service/addon.go:269\ngithub.com/oam-dev/kubevela/pkg/apiserver/interfaces/api.(*addonAPIInterface).listAddons\n\t/workspace/pkg/apiserver/interfaces/api/addon.go:129\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.8.0/filter.go:23\ngithub.com/oam-dev/kubevela/pkg/apiserver/domain/service.(*rbacServiceImpl).CheckPerm.func1\n\t/workspace/pkg/apiserver/domain/service/rbac.go:646\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.8.0/filter.go:21\ngithub.com/oam-dev/kubevela/pkg/apiserver/interfaces/api.authCheckFilter\n\t/workspace/pkg/apiserver/interfaces/api/authentication.go:123\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.8.0/filter.go:21\ngithub.com/oam-dev/kubevela/pkg/apiserver.(*restServer).requestLog\n\t/workspace/pkg/apiserver/server.go:257\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.8.0/filter.go:21\ngithub.com/oam-dev/kubevela/pkg/apiserver.(*restServer).OPTIONSFilter\n\t/workspace/pkg/apiserver/server.go:271\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.8.0/filter.go:21\ngithub.com/emicklei/go-restful/v3.(*Container).OPTIONSF

代理在etc、profile中弄的
<img width="883" alt="image" src="https://user-images.githubusercontent.com/24605302/199903815-7211aab6-4a3f-4757-9af8-60de354f31be.png">

**Comment by barnettZQG** [comment]
@liutao841204 这问题非常明显。你需要确保 `curl https://addons.kubevela.net/index.yaml` 可以正常请求。至于如何正确使用代理已经超出了 KubeVela 项目的范围，建议可以关闭该 Issue。

**Comment by liutao841204** [comment]
> @liutao841204 这问题非常明显。你需要确保 `curl https://addons.kubevela.net/index.yaml` 可以正常请求。至于如何正确使用代理已经超出了 KubeVela 项目的范围，建议可以关闭该 Issue。

真遗憾，看来我的环境不能用vela，我暂时放弃吧！

**Comment by wonderflow** [comment]
@liutao841204 Hi，针对这个问题，你可以自己搭一个内网的插件仓库来解决哈，不需要搭建网络代理  https://kubevela.io/zh/docs/platform-engineers/addon/addon-registry

**Comment by wonderflow** [comment, marked as answer]
内外环境下离线同步插件的文档可以参考： https://kubevela.io/zh/docs/platform-engineers/addon/addon-registry#%E5%9C%A8%E7%A6%BB%E7%BA%BF%E7%8E%AF%E5%A2%83%E4%B8%AD%E5%90%8C%E6%AD%A5-addon-%E5%88%B0-chartmuseum

**Reply by liutao841204** [reply]
收到，我在尝试一下

**Comment by liutao841204** [comment]
<img width="979" alt="image" src="https://user-images.githubusercontent.com/24605302/199947584-c236590a-527e-4f79-a670-76518f2013cd.png">
本地仓库可以，但我在添加仓库，还有几个报错，你们可以看看

**Reply by wonderflow** [reply]
ping @wangyikewxgm

**Comment by xueliangyang-oeuler** [comment]
@liutao841204  hello，catalog里的addon可能比较多，我们不一定要全部下载好，可以下载部分试着用起来，然后逐步增加。
